controllers/admin/AdminAjaxPs_buybuttonliteController.php

```diff
@@ -0,0 +1,118 @@
+
+        $sql = new DbQuery();
+        $sql->select('p.`id_product`, pa.`id_product_attribute`, pl.`name`, p.`reference`');
+        $sql->from('product', 'p');
+        $sql->join(Shop::addSqlAssociation('product', 'p'));
+        $sql->leftJoin('product_lang', 'pl', '
+            p.`id_product` = pl.`id_product`
+            AND pl.`id_lang` = ' . (int) $id_lang . Shop::addSqlRestrictionOnLang('pl')
+        );
+        $sql->leftJoin('product_attribute', 'pa', 'pa.`id_product` = p.`id_product`');
+        $where = 'pl.`name` LIKE \'%' . pSQL($query) . '%\'
+        OR p.`reference` LIKE \'%' . pSQL($query) . '%\'
+        OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_supplier` sp WHERE sp.`id_product` = p.`id_product` AND `product_supplier_reference` LIKE \'%' . pSQL($query) . '%\')';
+        $sql->orderBy('pl.`name` ASC');
+        if (Combination::isFeatureActive()) {
+            $where .= ' OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_attribute` `pa` WHERE pa.`id_product` = p.`id_product` AND (pa.`reference` LIKE \'%' . pSQL($query) . '%\'))';
+        }
+        $sql->where($where);
+        $result = Db::getInstance()->executeS($sql);
+        if (!$result) {
+            return false;
+        }
+        $results_array = array();
+        foreach ($result as $row) {
+            if ($this->hasCombinations($row['id_product'])) {
+                $row['attribute_name'] = $this->getAttributes($row['id_product_attribute'], $id_lang);
+                $row['image_link'] = $this->getProductAttributeImage($row['id_product'], $row['id_product_attribute']);
+            } else {
+                $row['image_link'] = $this->getProductImage($row['id_product']);
+            }
+            $results_array[] = $row;
+        }
+        $this->content = json_encode($results_array);
+    }
+
+    public function getProductImage($id_product)
+    {
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getCover($id_product);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function getProductAttributeImage($id_product, $id_product_attribute)
+    {
+        $context = Context::getContext();
+        $id_lang = $context->language->id;
+        $id_shop = $context->shop->id;
+
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getBestImageAttribute($id_shop, $id_lang, $id_product, $id_product_attribute);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function hasCombinations($id_product)
```

Comment:
phpDoc as always 😃 

It is not to bother you, it is a great asset for readability and can be used by IDEs to turn on features such as type validation, auto-complete ...